File: java/206/Solution.java
class Solution {
	public ListNode reverseList(ListNode head) {
		ListNode res = null, cur = head;
		while (cur != null) {
			ListNode next = cur.next;
			cur.next = res;
			res = cur;
			cur = next;
		}
		return res;
	}
}

File: cpp/206/solution.cpp
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};


class Solution {
    public:
        ListNode *reverseList(ListNode *head) {
            ListNode dummy, *cur = head;
            while (cur) {
                ListNode *next = cur->next;
                cur->next = dummy.next;
                dummy.next = cur;
                cur = next;
            }
            return dummy.next;
        }
};
